.github/workflows/publish.yml

```diff
@@ -23,7 +23,7 @@ jobs:
         run: python -m pip install build --user
 
       - name: Build a binary wheel and a source tarball
-        run: python -m build --sdist --wheel --outdir dist/ .
+        run: python -m build
```

Comment:
Explicit is better than implicit. I see no convincing reason not to pass the above arguments.

Reply:
That was quite tricky to explain why I opted to change it, but there is a subtle difference, and they explain it here: https://build.pypa.io/en/stable/#python--m-build


Option `--sdist` creates a source distribution (from the source directory).
Option `--wheel` creates a wheel (from the source directory).
Using either option of `--sdist`/`-s` or `--wheel`/`-w` disables the default behaviour. 
Up until now, it doesn't seem much.

So what is the default behaviour?
The default behaviour is to create both a source distribution, like `--sdist` does and, create a wheel, but instead of creating it from the source directory, it creates it from the source distribution.

This kinda gives you a little more assurance that what users get from a source distribution is equivalent to what they'd get from a wheel, or that a wheel is effectively able to be created from the source distribution.

Reply:
I see, thanks for the pointer.

However, I'm not sure why this design - logically, why shouldn't passing both flags do the same?
Also, it seems a bit problematic if there's no way to request a default explicitly?
But supposedly these are question for upstream.

Reply:
Did the previous solution it replaced work this way? If not, it's maybe that. But my attitude is more to use the most standardized config and commands as possible, so to use patterns that everyone uses, so it's a no brainer to maintain, always work as designed, and follow the ship of the good practices

Reply:
That said, I realize both behavior paths are needed, as building the wheel from the sdist might be problematic with some forms of VCS versioning.

e.g. for pypdfium2 it would work but add a dedicated suffix to indicate "built with given version file", which is useful to identify sdist installs, so `build`'s default may not really be desirable for us.
But it seems fine for ctypesgen.

Reply:
I don't really see the difference that it makes, as the version file is already in the source archive, as well in the wheel, and the PKG-INFO file has it evaluated.

How would you use that difference anyways? If the wheels aren't reproducible, that would be a bigger problem

Reply:
As mentioned, we use the difference to distinguish VCS installs from installs without VCS but given version file.

The wheels are reproducible from the VCS source, but not from the sdist, which in our case is intentional.
However I agree the drawback is we'd have to test the sdist separately.